### DISCUSSION TITLE
Long TOTP secret string not working

### DISCUSSION THREAD

**Discussion by stwongst** [keepassxreboot/keepassxc#9075] [Q&A]
Hello, we've been using KeePass* for long time without problem.   
Recently we deployed application systems that generate TOTP secret string with 32 characters long.  Though we can store the code in KeePassXC, the TOTP generated is incorrect.      For old accounts with TOTP secret string with 16 characters long, they worked probably without problem.    We're using KeePassXC 2.7.4.   Would anyone please help?

Thanks.

**Comment by droidmonkey** [comment]
Is the secret string still in base 32 format?

**Reply by stwongst** [reply]
I think yes, the string can be decoded using base32 decoder, and no error when pasting in KeepassXC TOTP setup page.  Thanks.

**Reply by droidmonkey** [reply]
Can you provide a sample secret from your program for our testing?

**Reply by stwongst** [reply]
Sorry, just found that the secret string comes with QR code generated by the application don't match...  Using the secret string in QR code without problem on KeePassXC.   Sorry for the trouble caused.
Thanks for your help.

**Comment by stwongst** [comment]
Hi, update:

Tried other applications that generates TOTP secret string of 32 bytes.   There is no problem with KeePassXC this time.
Probably it's the problem of the TOTP generating application...

Thanks.

**Reply by stwongst** [reply]
Update:  still not working if the secret string is in custom config : sha512, 8 digit code...
